CMakeLists.txt

```diff
@@ -12,13 +12,17 @@ set(CMAKE_CXX_STANDARD_REQUIRED ON)
 if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
   set(CMAKE_EXE_LINKER_FLAGS "-fuse-ld=gold -pthread")
 elseif(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
-  add_definitions(-DWINDOWS_IGNORE_PACKING_MISMATCH="") # vs2019 complains about using #pragma pack()
-  add_definitions(-DNOMINMAX) # please don't pull in these macros from <Windows.h>
-  add_definitions(-D_CRT_SECURE_NO_WARNINGS) # STL security warnings are just noise
-  add_definitions(-D_CRT_NONSTDC_NO_DEPRECATE) # POSIX deprecation warnings are also just noise
-  add_definitions(-D_USE_MATH_DEFINES) # Pull in M_PI and other <cmath> defines
+  add_compile_definitions(WINDOWS_IGNORE_PACKING_MISMATCH="") # vs2019 complains about using #pragma pack()
+  add_compile_definitions(NOMINMAX) # please don't pull in these macros from <Windows.h>
+  add_compile_definitions(_CRT_SECURE_NO_WARNINGS) # STL security warnings are just noise
+  add_compile_definitions(_CRT_NONSTDC_NO_DEPRECATE) # POSIX deprecation warnings are also just noise
+  add_compile_definitions(_USE_MATH_DEFINES) # Pull in M_PI and other <cmath> defines
   set(CMAKE_CXX_FLAGS "/MP ${CMAKE_CXX_FLAGS}")
   set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /SAFESEH:NO")
+elseif(CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
+  add_compile_options($<$<CONFIG:Debug>:-fsanitize=address>) # enable address sanitizer in debug builds
+  add_link_options($<$<CONFIG:Debug>:-fsanitize=address>)
```

Comment:
I would've prefer to have separate options for sanitizers and don't tie them to specific compiler/build type. If we even need them as such things easily could be passed with CFLAGS/CXXFLAGS/LDFLAGS to cmake on initial cache creation.

Reply:
Agreed.